### ISSUE TITLE
Cache issue on eavy load

### ISSUE THREAD

**Issue by ItSkary** [DannyBoyNg/UserAgentService#5]
Greetings and thanks for your project. 
I recently discovered that sometimes the parse method raise an exception, this happen expecially on heavy load : 

2022-10-26 11:50:12.674 +02:00 [FTL] Fatal error while procecssing https://my_site
System.ArgumentNullException: Value cannot be null.
Parameter name: key
   in Microsoft.Extensions.Caching.Memory.MemoryCache.TryGetValue(Object key, Object& result)
   in Microsoft.Extensions.Caching.Memory.CacheExtensions.GetOrCreate[TItem](IMemoryCache cache, Object key, Func`2 factory)
   in Ng.Services.UserAgentService.Parse(String userAgentString)
   in Project.Pages.BasePage.PropertiesInitialization() in C:\Users\myUser\source\repos\Project\Project\Pages\BasePage.aspx.cs:riga 299
   in Project.Pages.Products.PropertiesInitialization() in C:\Users\myUser\source\repos\Project\Project\Pages\Products.aspx.cs:riga 59
   in Project.Pages.BasePage.InitializeCulture() in C:\Users\myUser\source\repos\Project\Project\Pages\BasePage.aspx.cs:riga 190
   in ASP.pages_products_aspx.__BuildControlTree(pages_products_aspx __ctrl)
   in ASP.pages_products_aspx.FrameworkInitialize()
   in System.Web.UI.Page.ProcessRequest(Boolean includeStagesBeforeAsyncPoint, Boolean includeStagesAfterAsyncPoint)
   in System.Web.UI.Page.ProcessRequest()
   in System.Web.UI.Page.ProcessRequest(HttpContext context)
   in System.Web.HttpApplication.CallHandlerExecutionStep.System.Web.HttpApplication.IExecutionStep.Execute()
   in System.Web.HttpApplication.ExecuteStepImpl(IExecutionStep step)
   in System.Web.HttpApplication.ExecuteStep(IExecutionStep step, Boolean& completedSynchronously)

Here the code snippet that sometimes rise exception : 

```
        protected static readonly UserAgentSettings USER_AGENT_SETTINGS = new UserAgentSettings
        {
            CacheSizeLimit = 20000, 
            CacheSlidingExpiration = TimeSpan.FromDays(1),
            UaStringSizeLimit = 256
        };

        uaService = new UserAgentService(USER_AGENT_SETTINGS);
        this.UserAgent = uaService?.Parse(this.Request.UserAgent);
```

May be that i have to check the ASP.NET Request before calling the Parse method? At the moment i haven't checked if this.Request.UserAgent is just null or empty.

My project is an ASP.NET project written in C# with .Net 4.8. Your library version is the 1.0.3

**Comment by DannyBoyNg** [comment]
The issue is fixed with version 1.1.2

**Comment by ItSkary** [comment]
Thank a lot, just downloaded the updated version. 
Reading the code i am pretty sure the problem will be solved. Thanks again
